Adding command enter and handling for KeyDown and KeyPress

arraydude opened the pull request:
**Issue:** #240

**Decription:** 
- Adding command+enter for Mac users
- Adding test case
- Also handling ctrl in KeyDown

---

**Contribution License Agreement**

By submiting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

tvst commented on frontend/src/components/widgets/TextArea/TextArea.tsx:
```diff
@@ -69,29 +69,46 @@ class TextArea extends React.PureComponent<Props, State> {
+    this.handleSubmitEvent(e)
+  }
+
+  private onKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>): void => {
```
Why do we need both keydown and keypress? Won't this submit twice?

arraydude replied:
The metaKey is not being handled in the keyPress event

arraydude replied:
I mean the metaKey it's only captured within keyDown event

tvst replied:
I see... but the question remains: this will cause the text area to submit twice, right? If so, we should fix that.

arraydude replied:
We are preventing that using the 'dirty' check but i'll change it

jrhone commented on e2e/specs/text_area.spec.ts:
```diff
@@ -68,6 +68,20 @@ describe("st.text_area", () => {
     );
   });
 
+  it("sets value correctly on command-enter keypress", () => {
```
Was there a decision made to add support for `command-enter`? Not seeing it in the issue

arraydude replied:
I was thinking maybe the user tried to use command instead of control, it's a common ux issue

jrhone replied:
Gotcha, they said `ctrl-enter` in their issue comments. I'd maybe verify with them that they're using the correct combination before adding an extra one. We also tell them which keys to use in a helper message at the bottom right of the text area.

tvst replied:
I agree that we should check, just in case they _were_ using the `ctrl` key. Which would mean something else needs to be fixed.

That said, it makes sense to add support for `cmd-enter` too (even without updating the text area) since Mac users are so used to pressing `cmd` instead of `ctrl`.

Another solution would be so listen to `ctrl-enter` on Windows and Linux and `cmd-enter` on Mac. And update the message to match. If easy, then let's do this. Otherwise, the approach from this PR is fine too.

arraydude replied:
Done, now i'm checking the navigator.platform to show Command or Ctrl but in any case both would work

jrhone commented:
Text Area is working OK for me on develop. Unable to reproduce the issue the user is seeing. Anyone else?

arraydude commented:
> Ooops, I didn't mean to approve this!
> 
> The main thing to fix is the possible double-execution that happens when you press ctrl-enter on a text area.

The double-execution was catched using the dirty state but anyways i changed both events to be more secure and i also added the message depending on the platform

MastafaF commented:
Same problem in Chrome browser still to be handled. Works fine in Edge :)
